lib/fencing/st_client.c

```diff
@@ -2619,6 +2619,38 @@ stonith__sort_history(stonith_history_t *history)
     return new;
 }
 
+stonith_history_t *
+stonith__first_matching_event(stonith_history_t *history,
+                              bool (*matching_fn)(stonith_history_t *, void *),
+                              void *user_data)
+{
+    for (stonith_history_t *hp = history; hp; hp = hp->next) {
+        if (matching_fn(hp, user_data)) {
+            return hp;
+        }
+    }
+
+    return NULL;
+}
+
+bool
+stonith__event_state_pending(stonith_history_t *history, void *user_data)
+{
+    return history->state != st_failed && history->state != st_done;
+}
```

Comment:
We've been meaning to make this a library function for quite some time now ... good to have it somewhere :) There are a bunch of places where this could be useful, but that's something we can tackle separately